src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -153,6 +154,36 @@ public ResponseEntity<DatasetModel> retrieveDataset(
     return ResponseEntity.ok(datasetService.retrieveDatasetModel(id, userRequest, include));
   }
 
+  @Override
+  public ResponseEntity<SnapshotBuilderDatasetModel> retrieveSnapshotBuilderDataset(UUID id) {
```

Comment:
As I see this new endpoint in the wider Dataset API controller, I am realizing my preference that Cohort Builder endpoints be isolated to their own controller(s) to minimize confusion for both developers and users.  This would be similar to the approach that DUOS has taken when building endpoints meant solely for TDR to call:

https://consent.dsde-prod.broadinstitute.org/#/TDR

Separate cohort builder controllers could still call shared services if appropriate.

Reply:
My thinking is that this endpoint would be part of the datasets API, but it would be another `DatasetRequestAccessIncludeModel` flag.

Maybe the best way forward though is to create a more formal spec for the various endpoints we think we'll need, what paths they'll use, etc, based on the proposed API doc.